### DISCUSSION TITLE
edgex-ui troubles with Levski

### DISCUSSION THREAD

**Discussion by jdevoo** [edgexfoundry/community#122] [Q&A]
I recently updated a 2.1 docker-compose of edgex I use locally with docker-compose and no security.
After reviewing the docker-compose and launching the containers I am still facing issues with edgex-ui.
Somehow, only AppService is visible through the dashboard. All other services return 'Oops' e.g. system agent service is unavailable!
Consul shows all services are up and their logs look healthy. I am able to curl ping endpoints from exposed custom ports.
Below is my configuration.toml
```
[Writable]
LogLevel = 'INFO'

[Service]
Host = 'localhost'
Port = 4000
ServerBindAddr = ''
StartupMsg = 'edgex-ui-go service started'
HealthCheckInterval = '10s'
MaxRequestSize = 0
RequestTimeout = '5s'

[Clients]
    [Clients.CoreData]
    Protocol = 'http'
    Host = 'edgex-core-data'
    Port = 19880

    [Clients.Metadata]
    Protocol = 'http'
    Host = 'edgex-core-metadata'
    Port = 19881

    [Clients.CoreCommand]
    Protocol = 'http'
    Host = 'edgex-core-command'
    Port = 19882

    [Clients.Notification]
    Protocol = 'http'
    Host = 'edgex-support-notifications'
    Port = 19860

    [Clients.Scheduler]
    Protocol = 'http'
    Host = 'edgex-support-scheduler'
    Port = 19861

    [Clients.RuleEngine]
    Protocol = 'http'
    Host = 'edgex-kuiper'
    Port = 9081

    [Clients.System]
    Protocol = 'http'
    Host = 'edgex-sys-mgmt-agent'
    Port = 19890

    [Clients.AppService]
    Protocol = 'http'
    Host = 'edgex-app-rules-engine'
    Port = 19701

[Registry]
Host = 'host.docker.internal'
Port = 8500
Type = 'consul'
```
Below is the log for edgex-ui clicking on dashboard then on system agent. I tried setting INFO to DEBUG which doesn't add anything.
```
level=INFO ts=2023-06-04T19:08:24.099716167Z app=edgex-ui-service source=config.go:392 msg="Loaded service configuration from res/docker/configuration.toml"
level=INFO ts=2023-06-04T19:08:24.103829625Z app=edgex-ui-service source=variables.go:377 msg="Variables override of 'Service.Host' by environment variable: SERVICE_HOST=edgex-ui"
level=INFO ts=2023-06-04T19:08:24.104107497Z app=edgex-ui-service source=config.go:556 msg="Using local configuration from file (1 envVars overrides applied)"
level=INFO ts=2023-06-04T19:08:24.104533507Z app=edgex-ui-service source=httpserver.go:133 msg="Web server starting (edgex-ui:4000)"
level=INFO ts=2023-06-04T19:08:24.104568685Z app=edgex-ui-service source=message.go:50 msg="Service dependencies resolved..."
level=INFO ts=2023-06-04T19:08:24.104577479Z app=edgex-ui-service source=message.go:51 msg="Starting edgex-ui-service 2.3.0 "
level=INFO ts=2023-06-04T19:08:24.104583936Z app=edgex-ui-service source=message.go:55 msg="edgex-ui-go service started"
level=INFO ts=2023-06-04T19:08:24.104591877Z app=edgex-ui-service source=message.go:58 msg="Service started in: 12.820151ms"
/core-data/api/v2/ping
/core-metadata/api/v2/ping
/support-scheduler/api/v2/ping
/support-notifications/api/v2/ping
/sys-mgmt-agent/api/v2/ping
/sys-mgmt-agent/api/v2/ping
```
Any hint on troubleshooting this would be great. I used to have this work fine without reverse proxy. Happy to add if I can read more about its own configuration.
TIA

**Comment by jdevoo** [comment]
I added a reverse proxy. All good.

**Comment by bnevis-i** [comment]
Could not reproduce.  Did not have to add a reverse proxy to get edgex-ui to work.  Reverse proxy is not a supported feature in non-secure EdgeX.

**Reply by jdevoo** [reply]
bummer for me - that's how I had it before but couldn't figure out on my end.
Any chance you can share the configuration.toml? Any other consideration for the docker-compose.yaml itself? I guess the good news is that I am a step closer to finally turning on security properly 😛

**Reply by bnevis-i** [reply]
> bummer for me - that's how I had it before but couldn't figure out on my end. Any chance you can share the configuration.toml? Any other consideration for the docker-compose.yaml itself? I guess the good news is that I am a step closer to finally turning on security properly 😛

Literally all I did was (excusing some typos below now and again)

git clone https://github.com/edgexfoundry/edgex-compose
git checkout v2.1.0
cd edge-compose/compose-builder
make run no-secty
(wait a minute)
open up firefox to http://localhost:4000
